cli/cook/subcommands/show.py

```diff
@@ -208,63 +217,44 @@ def satisfy_pred():
-                if entity_type == 'job':
-                    wait_text = 'Waiting for the following jobs'
-                elif entity_type == 'instance':
-                    wait_text = 'Waiting for instances of the following jobs'
-                elif entity_type == 'group':
-                    wait_text = 'Waiting for the following job groups'
-                else:
-                    raise Exception('Invalid entity type %s.' % entity_type)
-
-                uuid_text = ', '.join([e['uuid'] for e in entities])
-                wait_text = '%s on %s: %s' % (wait_text, colors.bold(cluster['name']), uuid_text)
-                index = progress.add(wait_text)
-                if pred(entities):
-                    progress.update(index, colors.bold('Done'))
-                else:
-                    entities = wait_until(satisfy_pred, timeout, interval)
-                    if entities:
-                        progress.update(index, colors.bold('Done'))
-                    else:
-                        raise Exception('Timeout waiting for response.')
-            return entities
+    entities = http.make_data_request(lambda: make_request_fn(cluster, uuids))
+    if pred and len(entities) > 0:
+        if entity_type == 'job':
+            wait_text = 'Waiting for the following jobs'
+        elif entity_type == 'instance':
+            wait_text = 'Waiting for instances of the following jobs'
+        elif entity_type == 'group':
+            wait_text = 'Waiting for the following job groups'
         else:
-            return {}
-    except requests.exceptions.ConnectionError as ce:
-        logging.info(ce)
-        return {}
-    except json.decoder.JSONDecodeError as jde:
-        logging.exception(jde)
-        return {}
-
+            raise Exception('Invalid entity type %s.' % entity_type)
 
-def make_entity_request(cluster, endpoint, params):
-    """Attempts to query entities corresponding to the given params from cluster."""
-    url = make_url(cluster, endpoint)
-    resp = http.get(url, params=params)
-    logging.info('response from cook: %s' % resp.text)
-    return resp
+        uuid_text = ', '.join([e['uuid'] for e in entities])
+        wait_text = '%s on %s: %s' % (wait_text, colors.bold(cluster['name']), uuid_text)
+        index = progress.add(wait_text)
+        if pred(entities):
+            progress.update(index, colors.bold('Done'))
+        else:
+            entities = wait_until(satisfy_pred, timeout, interval)
+            if entities:
+                progress.update(index, colors.bold('Done'))
+            else:
+                raise Exception('Timeout waiting for response.')
+    return entities
 
 
 def make_job_request(cluster, uuids):
     """Attempts to query jobs corresponding to the given uuids from cluster."""
-    return make_entity_request(cluster, 'rawscheduler', params={'job': uuids, 'partial': 'true'})
+    return http.get(cluster, 'rawscheduler', params={'job': uuids, 'partial': 'true'})
```

Comment:
Wrap with `http.make_data_request`? (Ditto for immediately following calls.)

Reply:
These don't need it because they end up being passed as `make_request_fn` to `query_cluster`, which does the wrapping itself